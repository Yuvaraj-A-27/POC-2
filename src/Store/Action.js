import { ACTIVE_USER, ADD_TO_CART, CART_ACTIVE, CREATE_USER, CURRENT_PRODUCT, INITIAL_CATEGORY, INITIAL_PRODUCT, INITIAL_USER_DETAIL, LOGIN_ACTIVE, PRODUCT_POP_ACTIVE, PROFILE_ACTIVE, REGISTER_ACTIVE } from "./ActionType"

export const initialCategory = (value) =>{
    return{
        type: INITIAL_CATEGORY,
        payload : value
    }
}

export const initialProduct = (value) =>{
    return{
        type: INITIAL_PRODUCT,
        payload : value
    }
}

export const initialUserDetail = (value) =>{
    return{
        type:INITIAL_USER_DETAIL,
        payload : value
    }
}


export const loginActive = ()=>{
    return{
        type: LOGIN_ACTIVE,
    }
}

export const activeUser = (value)=>{
    return{
        type: ACTIVE_USER,
        payload: value
    }
}

export const productPopUpActive = ()=>{
    return{
        type: PRODUCT_POP_ACTIVE,
    }
}

export const currentProduct = (id) =>{
    return{
        type: CURRENT_PRODUCT,
        payload : id
    }
}

export const addToCart = (productId, userId) =>{
    return{
        type:ADD_TO_CART,
        payload: {
            productId,
            userId
        }
    }
}

export const profileActive = ()=>{
    return{
        type:PROFILE_ACTIVE,
    }
}

export const cartActive = () =>{
    return{
        type: CART_ACTIVE,
    }
}

export const registerActive = () =>{
    return{
        type : REGISTER_ACTIVE,
    }
}

export const createUser = (value) =>{
    return{
        type : CREATE_USER,
        payload : value
    }
}