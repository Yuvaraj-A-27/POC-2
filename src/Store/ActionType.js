export const INITIAL_CATEGORY = 'INITIAL_CATEGORY'
export const INITIAL_PRODUCT = 'INITIAL_PRODUCT'
export const INITIAL_USER_DETAIL = 'INITIAL_USER_DETAIL'
export const LOGIN_ACTIVE = 'LOGIN_ACTIVE'
export const ACTIVE_USER = 'ACTIVE_USER'
export const PRODUCT_POP_ACTIVE ="PRODUCT_POP_ACTIVE"
export const CURRENT_PRODUCT = 'CURRENT_PRODUCT'