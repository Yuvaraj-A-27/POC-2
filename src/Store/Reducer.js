import { ACTIVE_USER, ADD_TO_CART, CART_ACTIVE, CREATE_USER, CURRENT_PRODUCT, INITIAL_CATEGORY, INITIAL_PRODUCT, INITIAL_USER_DETAIL, LOGIN_ACTIVE, LOGIN_ERROR, LOGIN_ERROR_TYPING, PRODUCT_POP_ACTIVE, PROFILE_ACTIVE, REGISTER_ACTIVE } from "./ActionType";

const initialState ={
    category : [],
    product : [],
    userDetail : [],
    loginActive : false,
    productPopUpActive : false,
    profileActive : false,
    cartActive : false,
    registerActive : false,
    // loginError : false,
    activeUserDetail : '',
    currentProduct : 1,
    cart:[],
}

const Reducer = (state = initialState, action)=>{
    switch(action.type){
        case INITIAL_CATEGORY: return{
            ...state,
            category : [...action.payload]
        }

        case INITIAL_PRODUCT:return{
            ...state,
            product : [...action.payload]
        }

        case INITIAL_USER_DETAIL:return{
            ...state,
            userDetail : [...action.payload]
        }

        case LOGIN_ACTIVE: return{
            ...state,
            loginActive : !state.loginActive,
            productPopUpActive : false,
            profileActive:false,
            cartActive : false,
            registerActive : false,
        }

        case PRODUCT_POP_ACTIVE: return{
            ...state,
            productPopUpActive : !state.productPopUpActive,
            loginActive : false,
            profileActive:false,
            cartActive : false,
            registerActive : false,
        }

        case ACTIVE_USER: return{
            ...state,
            activeUserDetail : action.payload
        }

        case CURRENT_PRODUCT: return{
            ...state,
            currentProduct : action.payload
        }

        case ADD_TO_CART : return{
            ...state,
            cart : [...state.cart, action.payload]
        }

        case PROFILE_ACTIVE: return{
            ...state,
            profileActive : !state.profileActive,
            loginActive : false,
            productPopUpActive:false,
            cartActive : false,
            registerActive : false,
        }

        case CART_ACTIVE: return{
            ...state,
            cartActive : !state.cartActive,
            loginActive : false,
            productPopUpActive:false,
            profileActive : false,
            registerActive : false,
        }

        case REGISTER_ACTIVE : return{
            ...state,
            registerActive : !state.registerActive,
            cartActive : false,
            loginActive : false,
            productPopUpActive:false,
            profileActive : false
        }

        case CREATE_USER : return{
            ...state,
            userDetail : [...state.userDetail, action.payload]
        }

        // case LOGIN_ERROR : return{
        //     ...state,
        //     loginError : true
        // }

        // case LOGIN_ERROR_TYPING : return{
        //     ...state,
        //     loginError : false
        // }

        default : return state
    }
}

export default Reducer;