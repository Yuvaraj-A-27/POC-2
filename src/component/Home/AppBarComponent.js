import { AppBar, Button, IconButton, makeStyles, Toolbar, Typography } from '@material-ui/core';
import React from 'react';
import  MenuIcon from '@material-ui/icons/Menu'

const useStyles = makeStyles((theme) => ({
    root: {
      flexGrow: 1,
      marginTop : '-10px',
      top: 0,
    },
    menuButton: {
      marginRight: theme.spacing(16),
    },
    title: {
      flexGrow: 1,
    },
    btn:{
        padding:'0px 10px',
        '&:hover' :{
            backgroundColor:'red',
        }
    }
  }));

function AppBarComponent({id}){
    const classes = useStyles()


    return(
        <div>
            <AppBar id={id} position="sticky" className={classes.root}>
                <Toolbar>
                <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
                    <MenuIcon />
                </IconButton>
                <Typography variant="h5" className={classes.title}>
                    MyStore
                </Typography>
                <Button className = {classes.btn} color="inherit">Register</Button>
                <Button className = {classes.btn} color="inherit">Login</Button>
                </Toolbar>
            </AppBar>
        </div>
    )
}

export default AppBarComponent;