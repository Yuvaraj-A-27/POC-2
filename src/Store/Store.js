import { createStore } from "redux"
import persistReducer from "redux-persist/es/persistReducer"
import persistStore from "redux-persist/es/persistStore"
import Reducer from "./Reducer"
import storage from 'redux-persist/lib/storage'


const persistConfig = {
    key:'root',
    storage,
}

const persistorReducer = persistReducer(persistConfig, Reducer)

export const Store = createStore(persistorReducer)
export const Persistor = persistStore(Store)