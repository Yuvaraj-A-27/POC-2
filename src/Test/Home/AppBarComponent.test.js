import { fireEvent, render } from '@testing-library/react'
import AppBarComponent from '../../component/Home/AppBarComponent'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { LOGIN_ACTIVE, REGISTER_ACTIVE } from '../../Store/ActionType'

const middleware  = {}
const mockStore = configureStore(middleware)
const initialState = {}
const Store = mockStore(initialState)

describe('Home-AppBar component test',()=>{
    test('Home-AppBar component snapshot test',()=>{
        const AppBar = renderer.create(
            <Provider store={Store}>
                <AppBarComponent />
            </Provider>
        ).toJSON()
        expect(AppBar).toMatchSnapshot()
    })

    test('Home-AppBar component Functional testing',()=>{
        const {getByTestId} = render(
            <Provider store={Store}>
                <AppBarComponent />
            </Provider>
        )

        //action
        const registerActive = ()=>({type: REGISTER_ACTIVE})
        const loginActive = () => ({type: LOGIN_ACTIVE})


        const registerBtn = getByTestId('register-btn')
        const loginBtn = getByTestId('login-btn')

        // fireEvent.click(registerBtn)     
        Store.dispatch(registerActive())
                                                      //either fireEvent or Store.dispatch can be use to dispatch an action
        fireEvent.click(loginBtn)
        // Store.dispatch(LOGIN_ACTIVE())


        //checking dispatch of action
        expect(Store.getActions()).toEqual([{type:REGISTER_ACTIVE},{type:LOGIN_ACTIVE}])


    })
})