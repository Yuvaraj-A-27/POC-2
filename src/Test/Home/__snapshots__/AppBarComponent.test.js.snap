// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home-AppBar component test Home-AppBar component snapshot test 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary makeStyles-root-1 MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h5
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h5"
      >
        MyStore
      </h5>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-4 MuiButton-colorInherit"
        data-testid="register-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Register
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-4 MuiButton-colorInherit"
        data-testid="login-btn"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
    </div>
  </header>
</div>
`;
